src/app/components/ui/projectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+import Button from '../button';
+import Input from '../input';
+import useZodForm from '../../../hooks/useZodForm';
+import { useState } from 'react';
+import { z } from 'zod';
+import { addProject, editProject } from '../../../actions/actions';
+import { useRouter } from 'next/navigation';
+import { type Project } from '@prisma/client';
+//import init from 'zod-empty';
+//import ErrorContext from '../../../state/error/error.context';
+//import ErrorProvider from '../../../state/error/error.provider';
+
+interface Props {
+  project?: Project;
+}
+
+export const schema = z.object({
+  name: z
+    .string()
+    .nonempty("Please enter project name"),
+  upworkId: z.string(),
+  hubstaffId: z.string(),
+  asanaId: z.string(),
+});
+
+export default function ProjectForm({ project }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, isSubmitting, isDirty },
+  } = useZodForm({
+    schema,
+    //defaultValues: init(schema),
+    mode: 'onBlur',
+  });
+  const router = useRouter();
+  const [name, setName] = useState(project?.name ? project.name : "");
+  const [upworkId, setUpworkId] = useState(project?.upworkId ? project.upworkId : "");
+  const [hubstaffId, setHubstaffId] = useState(project?.hubstaffId ? project.hubstaffId : "");
+  const [asanaId, setAsanaId] = useState(project?.asanaId ? project.asanaId : "");
+
+  const onSubmit = handleSubmit((data) => {
+    if (project) {
+      const modifiedData = {
+        ...data,
+        id: project?.id,
+      }
+      editProject(modifiedData)
+      .catch(err => console.error('project edit error'))
+      .then(() => console.log('successfully edited'))
+      . catch(() => console.log('promise catched'));
```

Comment:
Why another `.catch()` after `.then()`? Your `.then()` callback is simple log and it can not throw - remove extra catch.